pkg/cmd/roachtest/main.go

```diff
@@ -334,13 +345,13 @@ func updateSpecForSelectiveTests(ctx context.Context, specs []registry.TestSpec)
 	}
 	fmt.Printf("%d selected out of %d successful tests.\n", numberOfTestsToSelect, len(successfulTests))
 	for i := range specs {
-		if testShouldBeSkipped(allTests, specs[i], roachtestflags.Suite) {
+		if testShouldBeSkipped(allTestsMap, specs[i], roachtestflags.Suite) {
 			specs[i].Skip = "test selector"
 			specs[i].SkipDetails = "test skipped because it is stable and selective-tests is set."
 		} else {
 			selectedTestsCount++
 		}
-		if td, ok := allTests[specs[i].Name]; ok {
+		if td, ok := allTestsMap[specs[i].Name]; ok && td != nil {
```

Comment:
What's the scenario that td can be nil? If we added a new test to the suite that doesn't have data in snowflake yet?

Reply:
yes